xdsl/traits.py

```diff
@@ -150,3 +151,21 @@ def verify(self, op: Operation) -> None:
                 f'Operation {op.name} must have a "sym_name" attribute of type '
                 f"`StringAttr` to conform to {SymbolOpInterface.__name__}"
             )
+
+
+class CallableOpInterface(OpTrait, abc.ABC):
+    """
+    Interface for function-like Operations that can be called in a generic way.
+
+    Please see MLIR documentation for CallOpInterface and CallableOpInterface for more
+    information.
+
+    https://mlir.llvm.org/docs/Interfaces/#callinterfaces
+    """
+
+    @classmethod
+    def get_callable_region(cls, op: Operation) -> Region:
```

Comment:
I guess we will add some argument handling here, right?

Reply:
(in a follow up PR)

Reply:
What do you mean?

Reply:
I was talking about the function arguments, but that is obv handled already as the arguments to the entry block 😅